modules/renter/contractor/downloader.go

```diff
@@ -109,11 +111,15 @@ func (c *Contractor) Downloader(id types.FileContractID) (Downloader, error) {
 	}
 	c.revising[contract.ID] = true
 	c.mu.Unlock()
-	releaseLock := func() {
-		c.mu.Lock()
-		delete(c.revising, contract.ID)
-		c.mu.Unlock()
-	}
+
+	// release lock early if function returns an error
+	defer func() {
+		if err != nil {
+			c.mu.Lock()
+			delete(c.revising, contract.ID)
+			c.mu.Unlock()
+		}
+	}()
```

Comment:
This code had not already been converted? I thought we made this change already.